tests/test/unsize.rs

```diff
@@ -142,6 +142,137 @@ fn dyn_to_dyn_unsizing() {
     }
 }
 
+#[test]
+fn super_auto_trait() {
+    test! {
+        program {
+            #[lang(unsize)]
+            trait Unsize<T> {}
+
+            #[object_safe]
+            trait Principal where Self: SuperAuto {}
+
+            #[auto]
+            #[object_safe]
+            trait SuperAuto {}
+
+            #[auto]
+            #[object_safe]
+            trait Auto {}
+        }
+
+        goal {
+            forall<'a> {
+                dyn Principal + 'a: Unsize<dyn Principal + SuperAuto + 'a>
+            }
+        } yields {
+            expect!["Unique; lifetime constraints [InEnvironment { environment: Env([]), goal: '!1_0: '!1_0 }]"]
+        }
+
+        goal {
+            forall<'a> {
+                dyn Principal + Auto + 'a: Unsize<dyn Principal + Auto + SuperAuto + 'a>
+            }
+        } yields {
+            expect!["Unique; lifetime constraints [InEnvironment { environment: Env([]), goal: '!1_0: '!1_0 }]"]
+        }
+    }
+}
+
+#[test]
```

Comment:
I feel like what's missing here is a test with the super trait being under a binder (e.g. `trait Principal where for<'a> Self: Super<'a>`. I *assume* where we can do this without modifying any of the test parsing code - if we can't, then I think it's okay to land this with just a FIXME added.

Reply:
does chalk represent `dyn for<'a> Trait<'a>`? I assume that `forall<'a> dyn Trait + 'a`  is a distinct thing from `dyn for<'a> Trait + 'a`. If it can represent that then it would probably also be good to test `dyn for<'a> Trait<'a>` can be upcasted to `dyn for<'a> Super<'a>` given some `trait Trait<'a>: Super<'a>` in addition to the case of `trait: for<'a> Super<'a>`

Reply:
I think it should be able to (there is certainly ir for it, but I'm not sure exactly how to specify it in the parser). If you can't figure it out, I'll look at it this evening.

That's a good test to add for sure.

Reply:
Test added: 5689335